premarum-backend/PreEnrollmentMgmt.WebApi/Controllers/StudentController.cs

```diff
@@ -0,0 +1,44 @@
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using PreEnrollmentMgmt.Core.Repositories;
+using PreEnrollmentMgmt.Core.Services;
+using PreEnrollmentMgmt.WebApi.Controllers.DTOS;
+
+namespace PreEnrollmentMgmt.WebApi.Controllers;
+
+
+[Route("api/[controller]")]
+[ApiController]
+
+public class StudentController : ControllerBase
+{
+    private readonly IMapper _mapper;
+    private readonly StudentService _studentService;
+    private readonly ITransactionManager _transactionManager;
+    
+    
+    public StudentController(ITransactionManager transactionManager, IMapper mapper,
+        StudentService studentService)
+    {
+        _transactionManager = transactionManager;
+        _mapper = mapper;
+        _studentService = studentService;
+    }
+
+    [HttpGet("Student/{studentEmail}")]
+    public async Task<IEnumerable<StudentDTO>> GetStudent([FromRoute] string studentEmail)
+    {
+        var student = await _studentService.GetStudent(studentEmail);
```

Comment:
When student first logs in it wont exist. You need to creat it if it does not exist. Consider renaming functions to GetOrCreateStudent